src/problems/triangle_LSJZ.clj

```diff
@@ -0,0 +1,92 @@
+(ns problems.triangle-LSJZ)
+
+; Solve the triangle puzzlie with search.
+
+
+;; Example of creating states
+;; start-state (tri/->State [[1] [1 1] [1 1 1] [1 1 1 1] [1 1 1 1 0]])
+;; goal-state (tri/->State [[0] [0 0] [1 0 0] [0 0 0 0] [0 0 0 0 0]])
+
+(defrecord State [board])
+
+(defn get-all-coords [state]
+   (apply concat
+          (map
+            (fn [index]
+              (map #(conj [index] %) (range (inc index))))
+            (range (count state)))))
+
+(def coords (get-all-coords [[][][][][]])) ;; Shhh, it's fine!
```

Comment:
All you care about in `get-all-coords` is `(count state)`, so why not just pass in that as a number (5 in this case)? I think that would be more readable, especially since you could then rename `state` to something meaningful like `number-levels`.